File: ruby/selection_sort.rb
def selection_sort(arr)
  sorted_arr = []
  while arr.length > 0
    min = arr.min
    index = arr.index(min)
    sorted_arr.push(min)
    arr.delete_at(index)
  end
  sorted_arr
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort([3, -1, 5, 2])

  puts

  # Don't forget to add your own!

  # BENCHMARK HERE, you can print the average runtime
  long_input = []

  100.times { long_input << rand }

  start_time = Time.now

  1000.times do 
    selection_sort([2, 1])
    selection_sort(long_input)
  end

  avg_time = (Time.now - start_time) / 1000
  puts avg_time
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/selection_sort.js
function selectionSort(arr) {
	const sortedArr = []
	while (arr.length > 0) {
		const min = Math.min(...arr)
		const index = arr.indexOf(min)
		sortedArr.push(min)
		arr.splice(index, 1)
	}
	return sortedArr
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSort([3, -1, 5, 2]));

  // BENCHMARK HERE, and print the average runtime
  const longInput = [];

  for (let i = 0; i < 100; ++i) {
    longInput.push(Math.random());
  }

  const startTime = Date.now();

  for (let i = 0; i < 1000; ++i) {
    selectionSort([2, 1]);
    selectionSort(longInput);
  }

  const avgTime = (Date.now() - startTime) / 2000;

  console.log(avgTime);
}

module.exports = selectionSort;

// Please add your pseudocode to this file
  /**************************************
   * declare an empty array called sortedArr
   * loop over arr while arr has elements
   *  find minimum value
   *  push it to sortedArr
   *  splice it from the original arr
   * return sortedArr
   */
// And a written explanation of your solution
